src/main/java/org/terasology/spawning/SpawnerComponent.java

```diff
@@ -45,10 +46,12 @@
     public boolean rangedSpawning;
 
     public int range = 20;
-    public int minDistance;
+    public int minDistance = 1;
     
     public boolean needsPlayer;
 
     public int playerNeedRange = 10000;
+
+    public Prefab requiredPrefab;
     
 }
```

Comment:
Part of the logic to get the `Prefab` to be spawned from the `PortalSystem` class itself. Added this line to the PortalSystem class `spawnerComponent.requiredPrefab = prefabManager.getPrefab("Oreons:OreonGuard");` to achieve this but this is still buggy.